### ISSUE TITLE
Control entropy for $tmpFileName (file name to long)

### ISSUE THREAD

**Issue by akomm** [Ocramius/GeneratedHydrator#86]
I just hit the roof of file (not path) length on my **fs** (encrypted ext4 = 143 bytes):

`AppHydrator__PM__ThingContainerYToxOntzOjc6ImZhY3RvcnkiO3M6NDE6IkdlbmVyYXRlZEh5ZHJhdG9yXEZhY3RvcnlcSHlkcmF0b3JGYWN0b3J5Ijt9.php.5d2dc0a9162c80.87561400`

Its 151 bytes, without namespaces.

While in production it will be 255 bytes, maybe a control over entropy is a good idea? Pass different `FileLocatorInterface` for `FileWriterGeneratorStrategy` doesn't really help as `uniqid` is not part of name generation process in the locator.

**Comment by Ocramius** [comment]
I suppose that we should use `tempnam()` instead

**Comment by akomm** [comment]
@Ocramius But you will probably still need to encode params to distinguish hydrators created with different params, right?

**Comment by Ocramius** [comment]
Yes, but it could be solved by using a hash and a signature on the hydrator. I do something similar on `ocramius/proxy-manager`

**Comment by akomm** [comment]
Its actually `CodeGenerationUtils`. `ocramius/proxy-manager` does not seem to use it. So you want to opt-out from `CodeGenerationUtils`, change it, or make different implementations here with the interfaces?

**Comment by flowl** [comment]
This is still a problem for windows users, especially while the namespace and everything is included in the resulting filename.

No issues on MacOS or Linux.
